### ISSUE TITLE
json: simplified Jackson support

### ISSUE THREAD

**Issue by elucash** [immutables/immutables#80]
After benchmarking and reevaluation (#68) It was decided to create greatly simplified annotation based Jackson integration without any special annotations from Immutables.

To integrate immutable object into Jackson, just add Jackson's annotation to delegate serialization/deserialization. It will trigger generation of minimal supporting infrastructure in immutable type itself.

```java
@Value.Immutable
@JsonSerialize(as = ImmutableConstructorJacksonMapped.class)
@JsonDeserialize(as = ImmutableConstructorJacksonMapped.class)
public interface ConstructorJacksonMapped {
  @JsonProperty("X") // customize field name using jacksons
  int instance();

  // this attribute will be automatically
  String bal();
}
```

Regular `ObjectMapper` could be used then to serialize and deserialize immutable objects. Integrated immutable objects could be mixed with "POJOs" in any kind of structure.

To support Guava's collections and `Optional` use `com.fasterxml.jackson.datatype:jackson-datatype-guava` artifact:

```java
ObjectMapper mapper = new ObjectMapper();
mapper.registerModule(new GuavaModule());
...
```

Implemented in 4bb0bee55a6685c43027e6d1a9c5a57d26da48ab

**Comment by elucash** [comment]
Code generation approach will be supported for #71.

**Comment by elucash** [comment]
Released in 2.0
